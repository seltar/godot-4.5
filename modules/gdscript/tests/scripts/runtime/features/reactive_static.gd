extends Node

@reactive
static var global_counter: int = 0

func test():
	@warning_ignore("return_value_discarded")
	global_counter_changed.connect(_on_global_counter_changed)
	
	# Test static property change
	global_counter = 42
	

func _on_global_counter_changed(old_value, new_value):
	print("Static global_counter changed: ", old_value, " -> ", new_value)
