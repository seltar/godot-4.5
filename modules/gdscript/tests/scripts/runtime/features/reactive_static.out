GDTEST_ANALYZER_ERROR
>> ERROR at line 3: "@reactive" annotation cannot be applied to a static variable.
>> ERROR at line 8: Identifier "global_counter_changed" not declared in the current scope.
